src/sentry/integrations/slack/utils/channel.py

```diff
@@ -103,16 +124,141 @@ def get_channel_id_with_timeout(
     integration: Integration | RpcIntegration,
     name: str | None,
     timeout: int,
-) -> tuple[str, str | None, bool]:
+) -> SlackChannelIdData:
+    """
+    Fetches the internal slack id of a channel using scheduled message.
+    :param integration: The slack integration
+    :param name: The name of the channel
+    :param timeout: Our self-imposed time limit.
+    :return: a SlackChannelIdData object
+    """
+
+    if name is None:
+        return SlackChannelIdData(prefix="", channel_id=None, timed_out=False)
+
+    _prefix = ""
+    _channel_id = None
+
+    time_to_quit = time.time() + timeout
+    client = SlackSdkClient(integration_id=integration.id)
+
+    logger_params = {
+        "integration_id": integration.id,
+        "name": name,
+    }
+    try:  # Check for channel
+        _channel_id = check_for_channel(client, name)
+        _prefix = "#"
+    except SlackApiError:
+        logger.exception("rule.slack.channel_check_error", extra=logger_params)
+        return SlackChannelIdData(prefix=_prefix, channel_id=None, timed_out=False)
+
+    if _channel_id is not None:
+        return SlackChannelIdData(prefix=_prefix, channel_id=_channel_id, timed_out=False)
+
+    return check_user_with_timeout(integration, name, time_to_quit)
+
+
+def check_user_with_timeout(
+    integration: Integration, name: str, time_to_quit: int
+) -> SlackChannelIdData:
+    """
+    If the channel is not found, we check if the name is a user.
+    """
+
+    _channel_id = None
+    _prefix = ""
```

Comment:
just wondering why you chose to have underscores here? these aren't global variables?

Reply:
i did that cause intellisense kept suggesting the variables defined in the previous function were in scope here.

Reply:
weird